### ISSUE TITLE
getSpoolContentById not converted with proper character encoding

### ISSUE THREAD

**Issue by JillieBeanSim** [zowe/zowe-explorer-vscode#2683]
**Describe the bug**

viewing spool files in Zowe Explorer jobs view is not converting with proper encoding
from what I see the zosmf cli call takes the session which should have the profile's encoding but doesn't seem to be used
`zowe.GetJobs.getSpoolContentById(this.getSession(), jobname, jobid, spoolId)`
looking at the zosmf rest api swagger the GET /records accepts a fileEncoding value
![Screenshot 2024-01-24 at 10 31 01 AM](https://github.com/zowe/vscode-extension-for-zowe/assets/49491949/0925a419-9108-4da8-8d5e-660fe8df9bae)

**Expected behavior**

profile encoding to be supported when downloading content of spool files and the file is converted correctly

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- Zowe Explorer Version:
- (Optional) Zowe CLI Version:
- (Optional) Are you using Secure Credential Store?

**Additional context**

**Comment by JillieBeanSim** [comment]
Depends on cli issue https://github.com/zowe/zowe-cli/issues/876

**Comment by t1m0thyj** [comment]
This is blocked by https://github.com/zowe/zowe-cli/pull/1822

We have already done the work on the Zowe client side to enable codepages for spool files, and have a ticket open with IBM regarding the z/OSMF encoding option not working as expected.

**Comment by awharn** [comment]
This is now unblocked as of Zowe CLI/SDKs 7.25.0

**Comment by t1m0thyj** [comment]
Resolved in https://github.com/zowe/zowe-explorer-vscode/pull/3361 - the `IJes.downloadSingleSpool` and `IJes.downloadSpoolContent` methods for z/OSMF API now support an encoding option, which is exposed as "Open with Encoding" option when right-clicking spool files in the Jobs tree.
